### ISSUE TITLE
Verify rDNS in app id is reachable

### ISSUE THREAD

**Issue by bbhtt** [flathub-infra/flatpak-builder-lint#379]
Additionally for code hosting domains check if the project exists.

If app id is `io.github.username.appname` check if github.com/username/appname exists.

This is to prevent wrong `usernames` being used in submissions.

This might cause some false positives as appname might not always match with repo name and sometimes people use code hosting app ids because they don't own a domain.

**Comment by hfiguiere** [comment]
existence isn't necessarily validation.

**Comment by bbhtt** [comment]
```
io.github.Dretch.MonomerFlatpakExample
io.github.Fndroid.clash_for_windows
io.github.Foldex.r2mod
io.github.Icalingua.Icalingua
io.github.Rangi42.polished-map-plusplus
io.github.RobertBeckebans.RBDoom3BFG-GL
io.github.Sebicodes99.Prt
io.github._0xzer0x.qurancompanion
io.github.aandrew_me.ytdn
io.github.arunsivaramanneo.GPUViewer
io.github.betaflight.BetaflightConfigurator
io.github.bytezz.IPLookup
io.github.carterbox.Electricity_Clock
io.github.congard.qnvsm
io.github.cudatext.CudaText-Qt
io.github.diegoivan.GaticTacToe
io.github.diegoivan.pdf_metadata_editor
io.github.dtsudo.ChessCompStompWithHacks
io.github.dtsudo.TuxPlanetSpeedrunAnyPercent
io.github.fabiangreffrath.Doom
io.github.fidele007.akkhara
io.github.fin_ger.OvGUCanteen
io.github.foxxmd.multiscrobbler
io.github.fralonra.PpdEditor
io.github.freedoom.FreeDM
io.github.freedoom.Phase1
io.github.freedoom.Phase2
io.github.garglk.Gargoyle
io.github.hakuneko.HakuNeko
io.github.hvdwofl.jExifToolGUI
io.github.idevecore.CurrencyConverter
io.github.lainsce.DotMatrix
io.github.libredirect.frontends-manager
io.github.martinrotter.rssguardlite
io.github.mimbrero.WhatsAppDesktop
io.github.mlm.asteroids_revenge
io.github.mmarco94.tambourine
io.github.nokse22.asciidraw
io.github.nokse22.minitext
io.github.paledega.alpine-rootfs
io.github.quodlibet.ExFalso
io.github.realmazharhussain.GdmSettings
io.github.rinigus.OSMScoutServer
io.github.rinigus.PureMaps
io.github.rocket.editor
io.github.sameboy.SameBoy
io.github.shiftey.Desktop
io.github.shiiion.primehack
io.github.sil_car.SqueezeVid
io.github.smolblackcat.Progress
io.github.srb2.rphys
io.github.subhra74.Muon
io.github.theforceengine.tfe
io.github.thiefmd.themegenerator
io.github.trigg.discover_overlay
io.github.trytonvanmeer.DungeonJournal
io.github.unrud.RecentFilter
io.github.weclaw1.ScoreTracker
io.github.xolatgames.marknpaul
io.gitlab.daikhan.stable
io.gitlab.jazztickets.choria
io.gitlab.jazztickets.emptyclip
io.gitlab.jazztickets.irrlamb
io.gitlab.news_flash.NewsFlash
io.gitlab.o20.word
io.gitlab.stone_kingdoms.StoneKingdoms
```

**Comment by bbhtt** [comment]
A few notes for code hosting app ids

- We check only ids with 3 components.
- For `io.github.username.appname` underscores in `username` must always be replaced by `-`. But people sometimes replace `_` with `-` in appname and sometimes they do not.

So we have to check two URLs, if both fails -> the project is not reachable. 

`https://github.com/{username_demangled}/{appname_orig}.git` and `https://github.com/{username_demangled}/{appname_demangled}.git`
